### PR TITLE
sql: add grammar help text for udf statements

### LINKED ISSUE #87425: sql: add help messages for UDF statements in `sql.y`
Add proper help messages for UDF statements in `sql.y` file. Also need to link online docs at the end of each help message when available.

Jira issue: CRDB-19343

### PR DESCRIPTION
**Pull request by chengxiong-ruan**
Backport resolves #87425

Release note: None
Release justification: necessary but low risk user experience change

### CODE REVIEW

File: pkg/sql/parser/sql.y [hunk 1]
```diff
@@ -1887,12 +1887,30 @@ alter_database_stmt:
...
+//    OWNER TO { new_owner | CURRENT_USER | SESSION_USER }
+// ALTER FUNCTION name [ ( [ [ argmode ] [ argname ] argtype [, ...] ] ) ]
+//    SET SCHEMA new_schema
+//
+// where action is one of:
+//
+//    CALLED ON NULL INPUT | RETURNS NULL ON NULL INPUT | STRICT
+//    IMMUTABLE | STABLE | VOLATILE
+//    [ NOT ] LEAKPROOF
 alter_func_stmt:
```
Comment (stbof) [inline pkg/sql/parser/sql.y:1907]:
```suggestion
// %SeeAlso: WEBDOCS/alter-function.html
alter_func_stmt:
```

File: pkg/sql/parser/sql.y [hunk 2]
```diff
@@ -4149,6 +4167,18 @@ create_extension_stmt:
...
+// CREATE [ OR REPLACE ] FUNCTION
+//    name ( [ [ argmode ] [ argname ] argtype [, ...] ] )
+//    [ RETURNS rettype ]
+//  { LANGUAGE lang_name
+//    | { IMMUTABLE | STABLE | VOLATILE }
+//    | [ NOT ] LEAKPROOF
+//    | { CALLED ON NULL INPUT | RETURNS NULL ON NULL INPUT | STRICT }
+//    | AS 'definition'
+//  } ...
 create_func_stmt:
```
Comment (stbof) [inline pkg/sql/parser/sql.y:4182]:
```suggestion
// %SeeAlso: WEBDOCS/create-function.html
create_func_stmt:
```

File: pkg/sql/parser/sql.y [hunk 3]
```diff
@@ -4400,6 +4431,11 @@ opt_routine_body:
     $$.val = (*tree.RoutineBody)(nil)
   }
 
+// %Help: DROP FUNCTION - remove a function
+// %Category: DDL
+// %Text:
+// DROP FUNCTION [ IF EXISTS ] name [ ( [ [ argmode ] [ argname ] argtype [, ...] ] ) ] [, ...]
+//    [ CASCADE | RESTRICT ]
 drop_func_stmt:
```
Comment (stbof) [inline pkg/sql/parser/sql.y:4439]:
```suggestion
// %SeeAlso: WEBDOCS/drop-function.html
drop_func_stmt:
```

**Review by ghost — CHANGES_REQUESTED** [review]

File: pkg/sql/parser/sql.y (hunk 1 above)
Comment (chengxiong-ruan) [reply pkg/sql/parser/sql.y:1908]: fixed as suggested

File: pkg/sql/parser/sql.y (hunk 2 above)
Comment (chengxiong-ruan) [reply pkg/sql/parser/sql.y:4184]: fixed as suggested

File: pkg/sql/parser/sql.y (hunk 3 above)
Comment (chengxiong-ruan) [reply pkg/sql/parser/sql.y:4442]: fixed as suggested

**Review by ghost — APPROVED** [review]

**Comment by chengxiong-ruan** [comment]
TFTR!
bors r+

**Pull request merged**